packages/common/src/controllers/CheckoutController.ts

```diff
@@ -42,11 +42,32 @@ export default class CheckoutController {
     this.subscriptionService = getNamedModule(SubscriptionService, integrationType);
   }
 
-  getSubscriptionOfferIds = () => {
-    return this.accountService.svodOfferIds;
+  initialiseOffers = async () => {
+    const mediaOffers = useCheckoutStore.getState().requestedMediaOffers;
+    const offersMedia = mediaOffers ? await this.getOffers({ offerIds: mediaOffers.map(({ offerId }) => offerId) }) : [];
+    useCheckoutStore.setState({ offersMedia });
+
+    if (!useCheckoutStore.getState().offersSubscription.length && this.accountService.svodOfferIds) {
+      const offersSubscription = await this.getOffers({ offerIds: this.accountService.svodOfferIds });
+      useCheckoutStore.setState({ offersSubscription });
+    }
+
+    if (!useCheckoutStore.getState().offersSwitchSubscription.length) {
+      const subscriptionSwitches = await this.getSubscriptionSwitches();
+      const offersSwitchSubscription = subscriptionSwitches ? await this.getOffers({ offerIds: subscriptionSwitches }) : [];
+      useCheckoutStore.setState({ offersSwitchSubscription });
+    }
+  };
+
+  getSubscriptionOfferIds = () => this.accountService.svodOfferIds;
+
+  chooseOffer = async (selectedOffer: Offer) => {
+    if (!selectedOffer) throw new FormValidationError({ form: [i18next.t('choose_offer.offer_not_found')] });
```

Comment:
The `selectedOffer` type suggest it can't be `null` or `undefined` making the early return redundant. Do we need to make it optional or remove the early return?

```suggestion
  chooseOffer = async (selectedOffer: Offer | null) => {
    if (!selectedOffer) throw new FormValidationError({ form: [i18next.t('choose_offer.offer_not_found')] });
```

Reply:
You're right, this should never happen.